### PR TITLE
Add Oracle Application Testing Suite exploit

### PR DESCRIPTION
**Pull request by wvu**
**What is it?**

> Application Testing Suite is a comprehensive, integrated testing solution that ensures the quality, scalability, and availability of your Web applications, Web Services, packaged Oracle Applications and Oracle databases.

Basically, it's an Oracle thing you use to test Oracle things (and more, presumably).

**How does it work?**

The vulnerability is a directory traversal that bypasses authentication and directly accesses the file upload form: `/olt/Login.do/../../olt/UploadFileUpload.do`. It's actually a pretty sweet one-liner, which is why I like it. You should be able to get a privileged shell on Windows.

**Verification steps:**
- [x] Download it here: http://www.oracle.com/technetwork/oem/downloads/index-084446.html
- [x] If you don't have a login, you can use this: http://bugmenot.com/view/oracle.com
- [x] Install the software and start it
- [x] Fire off the exploit and see if you get a shell

### CODE REVIEW

**Comment by wchen-r7** [comment]
It looks like the exploit is more specific for Windows, but it's in the multi directory?

File: modules/exploits/multi/http/oracle_ats_file_upload.rb [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    upload_jsp_shell
+    print_status("Executing JSP shell: #{full_uri}olt/pages/#{jsp_filename}")
+    exec_jsp_shell
+  end
+
+  # XXX: This is a hack because session.platform is a lie
+  def on_new_session(session)
+    super
+    print_status("Deleting JSP shell: #{jsp_path}")
+    session.shell_command_token("del /f /q #{jsp_path}")
```
Comment (wchen-r7) [inline modules/exploits/multi/http/oracle_ats_file_upload.rb]: This appears to be the same thing in FileDropper, which does a ```%Q|del.exe /f /q "#{win_file}"|```.... any reason that mixin isn't used?

Comment (wchen-r7) [inline modules/exploits/multi/http/oracle_ats_file_upload.rb]: Actually, I just saw the comment. Never mind. So I guess your session.platform is returning ```java``` or something like that, right?

Comment (wvu-r7) [inline modules/exploits/multi/http/oracle_ats_file_upload.rb]: ```linux```! Here's the problem: ``` 26       'Platform'      => %w{ linux osx solaris unix win },``` in ```modules/payloads/singles/java/jsp_shell_reverse_tcp.rb```.

**Comment by wvu-r7** [comment]
Oops, it was originally ```win``` and ```linux```, but I dropped it back down to ```win```. I'll add the Linux support back in, actually.

File: modules/exploits/multi/http/oracle_ats_file_upload.rb (hunk 1 above)
Comment (wvu-r7) [inline modules/exploits/multi/http/oracle_ats_file_upload.rb]: That should probably be ```java```.

**Comment by wchen-r7** [comment]
Ok, I'll add a Delayed while you work on it.

**Comment by wvu-r7** [comment]
Thanks, @wchen-r7!!

**Comment by bcook-r7** [comment]
Hi, any update for testing?

**Comment by bcook-r7** [comment]
I can't get the Oracle software to work, going to unassign for now.

**Comment by wwebb-r7** [comment]
I have a pretty good win/loss ratio against Oracle software, picking this up.

**Comment by wwebb-r7** [comment]
Merged, good job

**Comment by wvu-r7** [comment]
Thanks, was an easy one!

**Comment by tdoan-r7** [comment]
# Release Notes 

Oracle ATS Arbitrary File Upload - This module targets a directory traversal vulnerability in Oracle Testing Suite version 12.4.0.2.0 and earlier. It bypasses authentication and directly accesses the file upload form to upload and execute a JSP shell. Successful exploitation results in a privileged shell on Windows and Linux systems.

**Comment by wvu-r7** [comment]
@tdoan-r7: The privileged shell is on Windows only. :)

**Pull request merged**